doc/dev/lib.xml

```diff
@@ -12,6 +12,61 @@ kernel among the developers.
 
 <Chapter Label="Chap-Lib">
 <Heading>Maintaining the &GAP; kernel and library</Heading>
+
+<Section Label="Sect-debugging">
+<Heading>Debugging the GAP Kernel</Heading>
+
+The GAP kernel supports a variety of options which enabled when running <C>configure</C>,
+to help writing and debugging GAP kernel code.
+<P/>
+
+The first option, <C>--enable-debug</C>, is well supported can can be used
+whenever the GAP kernel, or kernel modules, are being developed (although
+developers should still check their code works without it).
+<P/>
+<C>--enable-debug</C> enables assertions throughout the kernel. These
+assertions are implemented using the <C>GAP_ASSERT</C> macro. These
+assertions are disabled when GAP is compiled without <C>--enable-debug</C>.
+When <C>--enable-debug</C> is enabled, <C>KernelDebug</C> will be shown in
+the Configuration line when GAP is started.
```

Comment:
Optional: Why is `Configuration` in upper case? If it is meant to "quote" the output of GAP, I'd mark it as such, e.g.
> the line starting with <C>Configuration:</C> when &GAP; is started.
Else, I'd use lower case. Same below.